### PR TITLE
Fixes for issue #108 & failing unit test fixture

### PR DESCRIPTION
**Pull request by pmonks**
* Fix for issue #108
* Fix for failing unit test fixture (URL had fallen out of sync with code)

Output from `clj -A:test -M:runner`:
```
$ clj -A:test -M:runner

Running tests in #{"test"}

Testing discljord.connections-test
May 12, 2022 5:43:14 PM clojure.tools.logging$eval10805$fn__10808 invoke
INFO: Connecting shard 0
2022-05-12 17:43:14.392:INFO::async-dispatch-3: Logging initialized @5522ms to org.eclipse.jetty.util.log.StdErrLog
May 12, 2022 5:43:15 PM clojure.tools.logging$eval10805$fn__10808 invoke
INFO: Full disconnect triggered from input

Ran 2 tests containing 11 assertions.
0 failures, 0 errors.
```

### CODE REVIEW

File: test/discljord/connections_test.clj [hunk 1]
```diff
@@ -77,7 +77,7 @@
                                                 "t" "READY"
                                                 "d" {"session_id" "session"}})))
                         nil)))]
-      (fake/with-fake-http ["https://discord.com/api/gateway/bot?v=8&encoding=json"
+      (fake/with-fake-http ["https://discord.com/api/gateway/bot?v=9&encoding=json"
```
Comment (IGJoshua) [inline test/discljord/connections_test.clj:80]: Could you change this to use the gateway api version var from discljord.http?

**Review by IGJoshua — CHANGES_REQUESTED** [review]
I like the upgrades, this looks good to me except for the http endpoint specifying a hard-coded version rather than using the api version var.

**Comment by pmonks** [comment]
Where is that car to be found?  Also, please let me know if I forgot to make this PR editable - I usually do that but sometimes forget.

**Review by JohnnyJayJay — CHANGES_REQUESTED** [review]
2 things:
- shouldn't the libraries be updated in leiningen as well?
- please add the version bumps to the changelog, under the "changed" section

File: test/discljord/connections_test.clj (hunk 1 above)
Comment (pmonks) [reply test/discljord/connections_test.clj]: Done.

**Comment by pmonks** [comment]
@JohnnyJayJay 
> 2 things:
> 
> - shouldn't the libraries be updated in leiningen as well?

Done.  See also issue #116.

> - please add the version bumps to the changelog, under the "changed" section

Done.

**Comment by pmonks** [comment]
Updated test output:
```
$ clj -A:test -M:runner
WARNING: Use of :main-opts with -A is deprecated. Use -M instead.

Running tests in #{"test"}

Testing discljord.connections-test
Jun 05, 2022 5:31:45 PM clojure.tools.logging$eval10967$fn__10970 invoke
INFO: Connecting shard 0
2022-06-05 17:31:45.468:INFO::async-dispatch-3: Logging initialized @5156ms to org.eclipse.jetty.util.log.StdErrLog
Jun 05, 2022 5:31:46 PM clojure.tools.logging$eval10967$fn__10970 invoke
INFO: Full disconnect triggered from input

Ran 2 tests containing 11 assertions.
0 failures, 0 errors.
```

**Review by JohnnyJayJay — APPROVED** [review]
looks good to me now, though I have changed the http-kit version back, we should use a stable version in discljord. If someone needs experimental things from the development version, they can include it in their own project.

**Comment by pmonks** [comment]
@JohnnyJayJay the stable version of http-kit (2.5.3) doesn't function correctly on any JVM newer than v11 (including the latest LTS, v17).  Therefore I would strongly encourage `discljord` to use http-kit v2.6.0-RC1, or alternatively wait until 2.6.0 final is released, while continuing to test with the RC(s).  The ongoing use of http-kit 2.5.3 is pretty problematic for anyone wanting to use an up to date JVM version.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Discljord follows semantic versioning.
 - Fix missing implementation for `add-channel-pinned-message!`, delegate to new endpoints `pin-`/`unpin-message`
 - Fix typo in `modify-guild-role!` endpoint name (`modifiy` -> `modify`)
 - Fix typo in `start-thread-without-message!` which prevented it from working
+- Fix issue #108 - dependencies have slowly fallen out of date
```
Comment (IGJoshua) [inline CHANGELOG.md:15]: Does Github automatically detect this and make a ref? If not could you make this a markdown link?

**Review by IGJoshua — COMMENTED** [review]
This looks great to me pending me figuring out if that reference to the issue in the changelog file will link to the issue in question.

**Review by IGJoshua — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (pmonks) [reply CHANGELOG.md]: It appears it doesn't, which is annoying given that it does elsewhere (e.g. in PRs).  No fixed.

**Comment by JohnnyJayJay** [comment]
> @JohnnyJayJay the stable version of http-kit (2.5.3) doesn't function correctly on any JVM newer than v11 (including the latest LTS, v17). Therefore I would strongly encourage `discljord` to use http-kit v2.6.0-RC1, or alternatively wait until 2.6.0 final is released, while continuing to test with the RC(s). The ongoing use of http-kit 2.5.3 is pretty problematic for anyone wanting to use an up to date JVM version.

I see. That is indeed unfortunate. I will confirm this and then chat about it with Joshua again. Note that if we leave it at 2.5.3, that doesn't mean you can't make a discljord project where you use a newer http-kit version, by simply excluding the transitive dependency from discljord and including your preferred version.

**Comment by pmonks** [comment]
FWIW http-kit now has an official 2.6.0 release out, so I'll look at updating this PR today, if that's helpful.

**Comment by pmonks** [comment]
Latest test output:
```shell
$ clj -A:test -M:runner

Running tests in #{"test"}

Testing discljord.connections-test
Sep 25, 2022 9:15:39 AM clojure.tools.logging$eval10967$fn__10970 invoke
INFO: Connecting shard 0
2022-09-25 09:15:39.392:INFO::async-dispatch-3: Logging initialized @4695ms to org.eclipse.jetty.util.log.StdErrLog
Sep 25, 2022 9:15:40 AM clojure.tools.logging$eval10967$fn__10970 invoke
INFO: Full disconnect triggered from input

Ran 2 tests containing 11 assertions.
0 failures, 0 errors.
```

**Comment by JohnnyJayJay** [comment]
> http-kit now has an official 2.6.0 release out
Seems that all we had to do was wait long enough for my concerns to vanish 😄

**Pull request merged**